framework/src/outputs/TableOutput.C

```diff
@@ -85,26 +99,48 @@ TableOutput::TableOutput(const InputParameters & parameters)
 void
 TableOutput::outputPostprocessors()
 {
-  // Add new row to the tables
-  if (_postprocessor_table.empty() ||
-      !MooseUtils::absoluteFuzzyEqual(
-          _postprocessor_table.getLastTime(), getOutputTime(), _new_row_tol))
-    _postprocessor_table.addRow(getOutputTime());
+  if (shouldOutputPostprocessorsRow(_postprocessor_table))
+    outputPostprocessorsRow(_postprocessor_table);
 
-  if (_all_data_table.empty() ||
-      !MooseUtils::absoluteFuzzyEqual(_all_data_table.getLastTime(), getOutputTime(), _new_row_tol))
-    _all_data_table.addRow(getOutputTime());
+  if (shouldOutputPostprocessorsRow(_all_data_table))
+    outputPostprocessorsRow(_all_data_table);
+}
 
-  // List of names of the postprocessors to output
-  const std::set<std::string> & out = getPostprocessorOutput();
+bool
+TableOutput::shouldOutputPostprocessorsRow(const FormattedTable & table)
+{
+  if (table.empty())
+    return true;
 
-  // Loop through the postprocessor names and extract the values from the PostprocessorData storage
-  for (const auto & out_name : out)
+  // Check time column; all options will output row if time is new
+  const Real old_time = table.getLastTime();
+  const Real new_time = getOutputTime();
+  if (!MooseUtils::absoluteFuzzyEqual(old_time, new_time, _new_row_tol))
+    return true;
+
+  // Check PP columns if that option enabled
+  if (_check_all_columns_for_new_row)
   {
-    const PostprocessorValue & value = _problem_ptr->getPostprocessorValueByName(out_name);
-    _postprocessor_table.addData(out_name, value);
-    _all_data_table.addData(out_name, value);
+    for (const auto & pp_name : getPostprocessorOutput())
+    {
+      const Real old_pp_value = table.getLastData(pp_name);
+      const Real new_pp_value = _problem_ptr->getPostprocessorValueByName(pp_name);
+      if (MooseUtils::absoluteFuzzyEqual(old_pp_value, new_pp_value, _new_row_tol))
+        return false;
```

Comment:
umm shouldnt you check all columns first before returning false?

Reply:
Looks correct to me: if *any* column is equal, we definitely should not output the row.